### ISSUE TITLE
Remove styles in header and footer and include them in a stylesheet? 

### ISSUE THREAD

**Issue by LittleHamster** [siteorigin/siteorigin-panels#367]
Thanks to @gregpriday the inline style attributes are now fixed. 
I was wondering if you guys can look into this issue as well, so I created a couple of rows and columns and this is added to the header: 
```
<style type="text/css" media="all" id="siteorigin-panels-grids-wp_head">
/* Layout 408 */ #pgc-408-0-0 , #pgc-408-0-1 { width:50% } #pg-408-0 .panel-grid-cell { float:left } #pl-408 .panel-grid-cell .so-panel { margin-bottom:30px } #pl-408 .panel-grid-cell .so-panel:last-child { margin-bottom:0px } #pg-408-0 { margin-left:-15px;margin-right:-15px } #pg-408-0 .panel-grid-cell { padding-left:15px;padding-right:15px } @media (max-width:780px){ #pg-408-0 .panel-grid-cell { float:none;width:auto } #pgc-408-0-0 { margin-bottom:30px } #pl-408 .panel-grid , #pl-408 .panel-grid-cell {  } #pl-408 .panel-grid .panel-grid-cell-empty { display:none } #pl-408 .panel-grid .panel-grid-cell-mobile-last { margin-bottom:0px }  } 
</style>
```

there is also one in the footer: 
```
<style type="text/css" media="all" id="siteorigin-panels-grids-wp_footer">
```

should these be loaded via enqueue style in a separate css file? isn't that more efficient?

**Comment by gregpriday** [comment]
Yes and no. The CSS generated by Page Builder is very specific to the page. External stylesheets are great when there's a lot of shared styling. So as your user browses your site, they always have the same stylesheet cached.

For single-use CSS like Page Builder generates, it's actually more efficient to have it inline because it reduces the need for for an additional HTTP request.

That said, we would like to add the option of having external stylesheets for people who want it, but it's not necessarily the best/only solution.

There shouldn't be styling in both your header and footer though. Assuming you're viewing a single page. I'll take a look at this.

**Comment by LittleHamster** [comment]
@gregpriday thanks for taking a look at this, it would be great to have an option to include all of these (all styles for different pages, posts) in a single style.css file.

**Comment by Rajik** [comment]
Any updates on this enhancement? We are in need of this feature and it would be great to know the timeline for this.

**Comment by gregpriday** [comment]
Going to close this issue for now. It's a great idea, but we've implemented a feature like this in the past but had to subsequently remove it because it caused so many issues and conflicts. Namely the content cache - https://github.com/siteorigin/siteorigin-panels/commit/afbc47b7ba6f307420303417f0abae3eaabe6186

For SiteOrigin Page Builder going forward, our goal is to keep things light weight and simple, and reduce the number of places our users can run into issues.
